### PR TITLE
Fix tf tree, add IMU frame, enable publishing only fisheye + depth

### PR DESCRIPTION
**Pull request by mfehr**

### CODE REVIEW

**Comment by mfehr** [comment]
test this please

**Comment by mfehr** [comment]
test this please

File: include/maplab-realsense/zr300-config.h [hunk 1]
```diff
@@ -56,6 +56,10 @@ struct ZR300Config {
   // IR config.
   bool infrared_enabled = true;
   int infrared_subsample_factor = 1;
+  bool infrared_auto_exposure = true;
+  int infrared_gain = 0.5;
```
Comment (schneith) [inline include/maplab-realsense/zr300-config.h]: int = 0.5 seems wrong

File: include/maplab-realsense/zr300-config.h [hunk 2]
```diff
@@ -56,6 +56,10 @@ struct ZR300Config {
...
+  int infrared_exposure = 0;
```
Comment (schneith) [inline include/maplab-realsense/zr300-config.h]: would it make sense to add a comment defining the valid range of values?

File: include/maplab-realsense/zr300-config.h (hunk 1 above)
Comment (schneith) [reply include/maplab-realsense/zr300-config.h]: would it make sense to add a comment defining the valid range of values?

File: include/maplab-realsense/zr300.h [hunk 3]
```diff
@@ -69,19 +69,26 @@ class ZR300 {
...
+
   ros::Publisher imu_publisher_;
   ros::Publisher pointcloud_publisher_;
 
-  rs::extrinsics T_infrared_fisheye_;
+  rs::extrinsics T_imu_fisheye_;
+  rs::extrinsics T_fisheye_fisheye_;
   sensor_msgs::CameraInfo fisheye_camera_info_;
-  rs::extrinsics T_infrared_depth_;
+  rs::extrinsics T_fisheye_depth_;
```
Comment (schneith) [inline include/maplab-realsense/zr300.h]: is the infrared is captured on the fisheye camera?

File: include/maplab-realsense/zr300.h [hunk 4]
```diff
@@ -69,19 +69,26 @@ class ZR300 {
...
+  // but pointclouds are enabled.
+  ros::Publisher color_camera_info_publisher_;
+  ros::Publisher depth_camera_info_publisher_;
+
   ros::Publisher imu_publisher_;
   ros::Publisher pointcloud_publisher_;
 
-  rs::extrinsics T_infrared_fisheye_;
+  rs::extrinsics T_imu_fisheye_;
+  rs::extrinsics T_fisheye_fisheye_;
```
Comment (schneith) [inline include/maplab-realsense/zr300.h]: is that always identity?

File: launch/maplab_realsense.launch [hunk 5]
```diff
@@ -34,7 +39,26 @@
...
     <param name="infrared/enabled" value="true"/>
     <param name="infrared/subsample_factor" value="1"/>
+    <param name="infrared/enable_auto_exposure" value="true"/>
+    <param name="infrared/emitter_enabled" value="true"/>
+    <!-- Values for outdoor calibration, disable autoexposure
+         for these settings to take effect. -->
+    <param name="infrared/gain" value="1"/>
+    <param name="infrared/exposure" value="5"/>
 
+<<<<<<< HEAD
```
Comment (schneith) [inline launch/maplab_realsense.launch]: git conflict

File: include/maplab-realsense/zr300.h (hunk 4 above)
Comment (schneith) [reply include/maplab-realsense/zr300.h]: rm if infrared==fisheye

File: src/zr300.cc [hunk 6]
```diff
@@ -204,9 +231,31 @@ void ZR300::retrieveCameraCalibrations() {
...
   };
+  auto getMotionExtrinsics = [](rs::stream stream, rs::device* zr300_device,
+                                rs::extrinsics* extrinsics) {
+    CHECK_NOTNULL(zr300_device);
+    CHECK_NOTNULL(extrinsics);
+    try {
+      *extrinsics = zr300_device->get_motion_extrinsics_from(stream);
+      LOG(INFO) << "Retrieved imu to camera extrinsics for " << stream
+                << "\n translation: " << extrinsics->translation[0] << ", "
+                << extrinsics->translation[3] << ", "
```
Comment (schneith) [inline src/zr300.cc]: index correct?

File: src/zr300.cc [hunk 7]
```diff
@@ -204,9 +231,31 @@ void ZR300::retrieveCameraCalibrations() {
...
+                << extrinsics->translation[2]
```
Comment (schneith) [inline src/zr300.cc]: index correct?

File: src/zr300.cc [hunk 8]
```diff
@@ -227,51 +276,65 @@ void ZR300::retrieveCameraCalibrations() {
     }
   };
 
+  getMotionExtrinsics(rs::stream::fisheye, zr300_device_, &T_imu_fisheye_);
+
   if (config_.fisheye_enabled) {
-    getExtrinsics(rs::stream::fisheye, zr300_device_, &T_infrared_fisheye_);
+    getExtrinsics(rs::stream::fisheye, zr300_device_, &T_fisheye_fisheye_);
```
Comment (schneith) [inline src/zr300.cc]: T_fisheye_fisheye ?

File: src/zr300.cc [hunk 9]
```diff
@@ -227,51 +276,65 @@ void ZR300::retrieveCameraCalibrations() {
...
     rs::intrinsics intrinsics_fisheye;
     getIntrinsics(rs::stream::fisheye, zr300_device_, &intrinsics_fisheye);
 
     convertCalibrationToCameraInfoMsg(
-        intrinsics_fisheye, T_infrared_fisheye_, &fisheye_camera_info_);
+        intrinsics_fisheye, T_fisheye_fisheye_, &fisheye_camera_info_);
```
Comment (schneith) [inline src/zr300.cc]: T_fisheye_fisheye?

File: src/zr300.cc [hunk 10]
```diff
@@ -306,25 +369,41 @@ geometry_msgs::TransformStamped ZR300::convertExtrinsicsToTf(
...
+  }
+
+  if (config_.infrared_enabled) {
+    extrinsics_transforms.push_back(convertExtrinsicsToTf(
+        T_fisheye_infrared_2_, stamp, parent, config_.kInfrared2Topic));
+    extrinsics_transforms.push_back(convertExtrinsicsToTf(
+        T_fisheye_infrared_, stamp, parent, config_.kInfraredTopic));
+  }
+
+  // NOTE: Should always be identity, since the fisheye frame is the frame of
```
Comment (schneith) [inline src/zr300.cc]: rm dead code?

**Review by schneith — APPROVED** [review]

File: launch/maplab_realsense.launch (hunk 5 above)
Comment (mfehr) [reply launch/maplab_realsense.launch]: Looks like github is lagging again, I fixed it and pushed it without this conflict..

**Comment by mfehr** [comment]
@schneith  thanks for the review

**Pull request merged**